tests/testMetadataObjectAccess.py

```diff
@@ -84,6 +85,8 @@ def test(self):
 
         self.assertIsInstance(visitInfo, lsst.afw.image.VisitInfo)
 
+        self.assertIsInstance(filt, lsst.afw.image.Filter)
```

Comment:
You could also check that the filter name is 'r'.  That would make sure the correct filter is being constructed.